src/backend/base/langflow/api/build.py

```diff
@@ -0,0 +1,443 @@
+            )
+            background_tasks.add_task(
+                telemetry_service.log_package_component,
+                ComponentPayload(
+                    component_name=vertex_id.split("-")[0],
+                    component_seconds=int(time.perf_counter() - start_time),
+                    component_success=valid,
+                    component_error_message=error_message,
+                ),
+            )
+        except Exception as exc:
+            background_tasks.add_task(
+                telemetry_service.log_package_component,
+                ComponentPayload(
+                    component_name=vertex_id.split("-")[0],
+                    component_seconds=int(time.perf_counter() - start_time),
+                    component_success=False,
+                    component_error_message=str(exc),
+                ),
+            )
+            logger.exception("Error building Component")
+            message = parse_exception(exc)
+            raise HTTPException(status_code=500, detail=message) from exc
+
+        return build_response
+
+    async def build_vertices(
+        vertex_id: str,
+        graph: Graph,
+        event_manager: EventManager,
+    ) -> None:
+        """Build vertices and handle their events.
+
+        Args:
+            vertex_id: The ID of the vertex to build
+            graph: The graph instance
+            client_consumed_queue: Queue for tracking client consumption
+            event_manager: Manager for handling events
+        """
+        build_task = asyncio.create_task(_build_vertex(vertex_id, graph, event_manager))
+        try:
+            await build_task
+            vertex_build_response: VertexBuildResponse = build_task.result()
+        except asyncio.CancelledError:
+            logger.debug("Build task cancelled")
+            build_task.cancel()
+            return
+
+        # send built event or error event
+        try:
+            vertex_build_response_json = vertex_build_response.model_dump_json()
+            build_data = json.loads(vertex_build_response_json)
+        except Exception as exc:
+            msg = f"Error serializing vertex build response: {exc}"
+            raise ValueError(msg) from exc
+
+        event_manager.on_end_vertex(data={"build_data": build_data})
+
+        if vertex_build_response.valid and vertex_build_response.next_vertices_ids:
+            tasks = []
+            for next_vertex_id in vertex_build_response.next_vertices_ids:
+                task = asyncio.create_task(
+                    build_vertices(
+                        next_vertex_id,
+                        graph,
+                        event_manager,
+                    )
+                )
+                tasks.append(task)
+            try:
+                await asyncio.gather(*tasks)
+            except asyncio.CancelledError:
+                for task in tasks:
+                    task.cancel()
+                return
```

Comment:
Not needed. The cancel request will be propagated to the tasks
```suggestion
            await asyncio.gather(*tasks)
```